flutter/lib/ui/history/result_list_screen.dart

```diff
@@ -54,23 +99,62 @@ class _ResultListScreenState extends State<ResultListScreen> {
         separatorBuilder: (context, index) => const Divider(),
         itemBuilder: (context, index) {
           final item = itemList[index];
-          return ListTile(
-            title: Text(
-              helper.formatDate(item.meta.creationDate.toLocal()),
-              style: const TextStyle(fontWeight: FontWeight.bold),
-            ),
-            subtitle: Text(item.meta.uuid),
-            onTap: () {
+          return item.build(context);
+        },
+      ),
+    );
+  }
+
+  List<ListItem> listItems(List<ExtendedResult> items, ResultFilter filter) {
+    // Filter and sort the items based on the selected filter option
+    List<ExtendedResult> filteredItems = List<ExtendedResult>.from(items);
+
+    if (filter.sortBy == SortBy.taskThroughputDesc) {
+      return _benchmarkResults(filteredItems, filter);
+    }
+    return _extendedResults(filteredItems, filter);
+  }
+
+  List<ListItem> _benchmarkResults(
+      List<ExtendedResult> items, ResultFilter filter) {
+    List<BenchmarkExportResult> benchmarks = items
+        .expand((item) => item.results)
+        .where((element) =>
+            element.performanceRun != null &&
+            element.performanceRun!.throughput != null)
+        .toList();
+
+    benchmarks.sort((a, b) =>
+        b.performanceRun!.throughput!.compareTo(a.performanceRun!.throughput!));
+
+    return benchmarks
+        .map((benchmark) => BenchmarkListItem(benchmark, () {
+              Navigator.push(
+                context,
+                MaterialPageRoute(
+                  builder: (context) => RunDetailsScreen(result: benchmark),
+                ),
+              ).then((value) => setState(() {}));
+            }))
+        .toList();
+  }
+
+  List<ListItem> _extendedResults(
+      List<ExtendedResult> items, ResultFilter filter) {
+    if (filter.sortBy == SortBy.dateAsc) {
+      items.sort((a, b) => a.meta.creationDate.compareTo(b.meta.creationDate));
+    } else if (filter.sortBy == SortBy.dateDesc) {
+      items.sort((a, b) => b.meta.creationDate.compareTo(a.meta.creationDate));
+    }
```

Comment:
Can you extract all the sorting logics to a separated class, similar to how `class ResultFilter` works?
That makes it easier to extend the sorting options in the feature.

Reply:
Refactored